processor/src/lib.rs

```diff
@@ -64,6 +68,17 @@ pub fn execute(script: &Script, inputs: &ProgramInputs) -> Result<ExecutionTrace
     Ok(ExecutionTrace::new(process))
 }
 
+/// Returns an execution iterator that allows callers to step through each exceution and inspect
+/// vm state information along side.
+pub fn execute_iter(
+    script: &Script,
+    inputs: &ProgramInputs,
+) -> Result<VmStateIterator, ExecutionError> {
+    let mut process = Process::new(inputs.clone());
+    process.execute_code_block(script.root())?;
+    Ok(VmStateIterator::new(process))
+}
```

Comment:
I'm thinking we should catch an execution error inside this function and always return an iterator. So, basically, even if there is an error, we should return an iterator which would allow iterating over the states before the error happens.